### ISSUE TITLE
Dark mode for admin area 

### ISSUE THREAD

**Issue by dignajar** [bludit/bludit#987]
The admin area use bootstrap, maybe is possible to download a bootstrap style in dark and switch between them.

**Comment by Joduai** [comment]
That would be a nice and simple to add feature.
The more, couple additional bootswatch css than darkly could be added.
I'm mostly into slate / superhero themes, as they are similar to color sets I use in code editors ;)

But once there's an option to change basically css file, adding more manually would be blazing fast.

**Comment by philippd1** [comment]
any idea on how we might implement this? I would like to work on this one.
Also, really like @Joduai 's idea of using bootswatch themes, would indeed be a easy way of theming

**Comment by ltGuillaume** [comment]
I was pleasantly surprised by the result of simply putting two lines of CSS in for my Music Folder Player:
```
body { filter: invert(94%); }
img { filter: invert(94%); } /* don't invert images */
```

**Comment by philippd1** [comment]
while I definitely get your point and this way of solving the request, I think we are actually talking about a proper adminTheme Support in this thread.
I'm actually working on this and have got a few samples here (using darkly by bootswatch at the moment)
![image](https://user-images.githubusercontent.com/17651032/54444086-528e9380-4742-11e9-9a5a-a1a1d68ceac2.png)
![image](https://user-images.githubusercontent.com/17651032/54444197-8669b900-4742-11e9-9ac9-5c133d46d94b.png)

**Comment by BlakesHeaven** [comment]
I quite like a striking black design for pictures and support the option of more choice in theme colours. From a design point of view, can  I  suggest you read  through this web page?
[Websites with dark backgrounds: How to do it properly](https://undsgn.com/website-dark-backgrounds/)

In it they say:
> The main concern of dark designs is readability, and if it is not done right, the visitors and even your client may find it unappealing. With this in mind, you should pay extra close attention to the text – its format, style, font and compatibility with the background color.

They go on to suggest:

> We suggest increasing the white space in the text by adjusting paragraph size, kerning, and leading, as they will certainly improve readability.

Perhaps a style switcher setting should be included in any implementation.
Thanks for listening 🙂

**Comment by philippd1** [comment]
![image](https://user-images.githubusercontent.com/17651032/54696421-14231b00-4b2c-11e9-8970-b0e8fd8fbba3.png)
Okay, so I just built a plugin for embedding a custom stylesheet and a custom html style tag.
I personally recommend themes from https://bootswatch.com/ , but you still might have to tweak them using your own custom css

**Comment by philippd1** [comment]
https://github.com/philippd1/admin-theme
the plugin is now available 😉

**Comment by ltGuillaume** [comment]
Your `EOF;` are indented. Apparently you shouldn't do that (see http://php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc).

**Comment by philippd1** [comment]
![image](https://user-images.githubusercontent.com/17651032/54704090-679c6580-4b3a-11e9-94ea-49ccf544dcfc.png)
Bootswatch Theme API integration now done ✔
You can simply click on "use theme" and the selected theme gets set

**Comment by philippd1** [comment]
> Your `EOF;`s are indented. Apparently you shouldn't do that (see http://php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc).

yeah, I actually don't really like the use of EOF... now working with single quotes 😉

**Comment by Joduai** [comment]
@philippd1 
> the plugin is now available 😉

I looked at the repo and I wondered why there's no assets dir.
After checking content plugin.php I am a bit disappointed. 
Your solution links to external files, and this totally stops me from using your plugin.
@dignajar explicitly mentions on homepage, that bludit doesn't utilize external libs nor assets.
Imo you should at least let the user know about this.

**Comment by philippd1** [comment]
so you're telling me that I should download all of the assets from bootswatch and ship everything with the plugin?

**Comment by Joduai** [comment]
You could come up with both option: loading external assets or downloading and saving theme files on the server.
Of course assuming that second option doesn't open any security hole. 
From the other hand if user's hosting account has disabled php file operation functions by default, he could still use external option.

Having on mind that additional "plugin installation" plugin is on the way, a complete zip package is going to be relatively small (1 img+1 css file per every theme).  Of course comparing to unzipped css files , which would take ~4MB and has to be send through ftp.

**Comment by philippd1** [comment]
alright, I will rethink this and work on it when I have some more time on my hand.

**Comment by dignajar** [comment]
I feel is need to polish a little the current CSS `bludit.css`, because I overwrites some styles from Bootstrap.

This file overwrites the styles.
https://github.com/bludit/bludit/blob/master/bl-kernel/admin/themes/booty/css/bludit.css

Would be good to use some default styles from Bootstrap so is easier to change just the bootstrap theme and the colors, for example the textareas and text inputs the background is white which is not good.

**Comment by dignajar** [comment]
At the end what I want is switch between CSS files, not change the HTML code.

**Comment by philippd1** [comment]
> At the end what I want is switch between CSS files, not change the HTML code.

yeah, I get that.
As you said tho, this is not possible with the current bludit css overwriting Bootstrap styles.

**Comment by Kazem-ma79** [comment]
> ![image](https://user-images.githubusercontent.com/17651032/54704090-679c6580-4b3a-11e9-94ea-49ccf544dcfc.png)
> Bootswatch Theme API integration now done ✔
> You can simply click on "use theme" and the selected theme gets set

It's good but can't back to stock theme and it would be better to use full theme not just style file.

**Comment by philippd1** [comment]
well..... if you would've read the thread carefully (😅 u would know
- that this plugin currently isn't actively maintained
- and that theming is not as easy as you'd think/ hope because of internal bootstrap overrides of the default Bludit admin theme (booty)

as for not being able to revert to the stock Bludit Theme: I always welcome PRs on the repo :)
https://github.com/philippd1/bludit-admin-theme

**Comment by Kazem-ma79** [comment]
> well..... if you would've read the thread carefully (😅 u would know
> 
> * that this plugin currently isn't actively maintained
> * and that theming is not as easy as you'd think/ hope because of internal bootstrap overrides of the default Bludit admin theme (booty)
> 
> as for not being able to revert to the stock Bludit Theme: I always welcome PRs on the repo :)
> https://github.com/philippd1/bludit-admin-theme

Yes but I'm making some themes for admin panel, didn't seen any problems yet (Except activating them from admin panel) 😁.
